sdk/storage/src/blob/container/requests/list_blobs_builder.rs

```diff
@@ -132,9 +132,8 @@ impl<'a> ListBlobsBuilder<'a> {
             .await?;
 
         Ok(ListBlobsResponse::from_response(
-            self.container_client.container_name(),
             response.headers(),
-            &std::str::from_utf8(response.body())?,
+            &std::str::from_utf8(&response.body()[3..])?,
```

Comment:
Why a slice from 3 on? Not only does this raise questions (maybe at least a content), what's with the magic number?

Reply:
It's for skipping the BOM ([https://www.opentag.com/xfaq_enc.htm#enc_bom](https://www.opentag.com/xfaq_enc.htm#enc_bom)). For UTF-8 is optional but it seems that `serde_xml` doesn't support it so I just forcibly remove from the input stream.

It's probably better to have a function that looks for the BOM before removing it: while I am confident Azure will always return it, an empty string would cause this code to panic.

Reply:
Yeah, eventually checking for it is probably good. I agree it's incredibly unlikely, but unless it's documented (and even then, it just makes it less likely but still possible) I wouldn't count on it too much. Given this code, funny how the absence of a BOM would make the code blow up. 😉

Reply:
Done! I've also made the response more aligned to the other responses by implementing `std::convert::TryFrom`.